### DISCUSSION TITLE
JavaMelody Confluence access without Admin rights

### DISCUSSION THREAD

**Discussion by knaflitsch** [javamelody/javamelody#991] [Q&A] [answered]
Hi,

we have installed the JavaMelody Plugin on our on-premises Confluence application. However only admins have access to the monitoring and we want to grant only very very few people admin rights. 

We are starting to monitor our Confluence better since it's getting bigger and bigger and want to compare  our End-to-end monitoring KPIs with KPIs from JavaMelody (for example response time and user sessions).

For this we need to grant our monitoring team access to JavaMelody without granting them ConfluenceAdmin rights, so they can extract the data via the API. 
Is it possible to grant others access?

Thank you!
![image](https://user-images.githubusercontent.com/12643023/105834677-10376e00-5fcb-11eb-8052-1b7b455346cc.png)

**Comment by evernat** [comment, marked as answer]
Hi Elija,

The users authorized to use the plugin are not configured one by one. So the plugin needs a way to know which users are authorized. There is a Confluence api which seems right for that: [isConfluenceAdministrator](https://docs.atlassian.com/atlassian-confluence/6.6.0/com/atlassian/confluence/security/PermissionManager.html#isConfluenceAdministrator-com.atlassian.user.User-).

But I do not see a Confluence api like isMonitoringUser for your different need, so what you ask is not possible at the moment, Sorry.

Otherwise there is a radical alternative for you: disable the Confluence auth for the plugin and use your configured usernames and passwords. For that, you can add for example in `bin/setenv.sh` (if Linux):
```sh
CATALINA_OPTS="-Djavamelody.plugin-authentication-disabled=true -Djavamelody.authorized-users=user1:password1,user2:password2 ${CATALINA_OPTS}"
```
Passwords can be hashed, see the end of the chapter at https://github.com/javamelody/javamelody/wiki/UserGuide#16-security
Then all users of the plugin will be asked for basic auth without a need to be authentified as a Confluence user.

By the way, what you ask was just added to the soon to be released Jenkins monitoring plugin (v1.87), with the Overall/SystemRead permission which is possible beside the Administer permission. It seems like a good idea.
